demos/action_recognition_demo/python/action_recognition_demo.py

```diff
@@ -97,23 +98,27 @@ def main():
 
     encoder_xml = args.m_encoder
     encoder_bin = args.m_encoder.replace('.xml', '.bin')
-    encoder = IEModel(encoder_xml, encoder_bin, ie, encoder_target_device,
-                      num_requests=(3 if args.device == 'MYRIAD' else 1))
-
-    if args.m_decoder is not None:
+    models = [IEModel(encoder_xml, encoder_bin, ie, encoder_target_device,
+                num_requests=(3 if args.device == 'MYRIAD' else 1))]
+
+    if args.architecture_type == 'dummy-de':
```

Comment:
IMO, naming an architecture after what it _doesn't_ have is not the best strategy. I would suggest `en-mean` instead, which describes how the model works (you encode every frame and then you take the mean).

Reply:
Yes, I agree, `en-mean` would be better, thanks.

Reply:
Updated